### PR TITLE
Added design doc on randomized testing.

### PR DESCRIPTION
**Pull request by mwhittaker**

### CODE REVIEW

File: docs/randomized_testing.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
+# Randomized Testing
+
+When you write a **unit test**, you write some code and check that it executes
```
Comment (ghemawat) [inline docs/randomized_testing.md:3]: I love this introduction. Very well written and to the point.

File: docs/randomized_testing.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+In this document, we describe three ways to introduce randomized property-based
+testing to Service Weaver. This doc borrows heavily from [*Testing Distributed
+Systems*][testing_distributed_systems] by [Andrey Satarin][asatarin].
+
+The various approaches differ in the following ways.
+
+- What properties can be tested?
+- What kind of failures are supported?
+- How replayable, readable, and minimizable are failing executions?
```
Comment (ghemawat) [inline docs/randomized_testing.md:50]: Maybe s/minimizable/minimal/
?

**Review by ghemawat — APPROVED** [review]
Great doc!

File: docs/randomized_testing.md (hunk 2 above)
Comment (mwhittaker) [reply docs/randomized_testing.md]: Done.

**Pull request merged**